### PR TITLE
EU countries: EU VIES.

### PR DESCRIPTION
**Pull request by coquer**

### CODE REVIEW

File: src/CompanyService.php [hunk 1]
```diff
@@ -6,11 +6,13 @@
 use Ageras\Sherlock\Exceptions\SingleResultExpected;
 use Ageras\Sherlock\Providers\CompanyProviderInterface;
 use Ageras\Sherlock\Providers\CvrProvider;
+use Ageras\Sherlock\Providers\EUProvider;
 
 class CompanyService
 {
     protected $providers = [
         'dk' => CvrProvider::class,
+        'nl' => EUProvider::class,
```
Comment (coquer) [inline src/CompanyService.php:15]: After testing we need to add the other countries here :) @phidah

File: src/Providers/EUProvider.php [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    }
+
+    /**
+     * Format address.
+     * @param $data
+     * @return null|string
+     */
+    protected function formatAddress($data)
+    {
+        $datas = explode("\n", $data);
```
Comment (phidah) [inline src/Providers/EUProvider.php]: Could be done in a smoother way.

File: src/Providers/EUProvider.php [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    }
+
+    /**
+     * Format city.
+     * @param $data
+     * @return null|string
+     */
+    protected function formatCity($data)
+    {
+        $data = explode(' ', $data);
```
Comment (phidah) [inline src/Providers/EUProvider.php]: This assumes that the city is the last element, which might not be true in all cases.

File: src/Providers/EUProvider.php [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+        return isset($city) ? trim($city) : null;
+    }
+
+    /**
+     * Format postcode.
+     * @param $data
+     * @return null
+     */
+    protected function formatPostcode($data)
```
Comment (phidah) [inline src/Providers/EUProvider.php]: Move all parsing to a common method and simplify.

File: src/Providers/EUProvider.php (hunk 2 above)
Comment (coquer) [inline src/Providers/EUProvider.php]: Im not happy with one either 👍 but they return`""" "" "" \n address """ \n` so I need to tested to get proper pattern and maybe regx? what do you think is the best way to do it?

**Pull request merged**